build/datagen-steps/metrics.js

```diff
@@ -0,0 +1,425 @@
+    );
+  }
+  // Write metric file
+  const dest = path.join(outputBase, geography);
+  return writeMetricFile(dest, metric, outJSON);
+}
+
+/**
+ * Convert weighted metric from CSV to JSON and write out resulting file(s).
+ *
+ * Can throw errors.
+ */
+async function convertWeightedMetric(geography, metric, { inputBase, outputBase }) {
+  const outJSON = {};
+  const files = [
+    await checkMetricFileName(geography, metric, "r", inputBase),
+    await checkMetricFileName(geography, metric, "d", inputBase),
+  ];
+
+  if (!files[0]) {
+    throw Error(`Could not find numerator (R) CSV file for weighted metric.`);
+  }
+
+  if (!files[1]) {
+    throw Error(`Could not find denominator (D) CSV file for weighted metric.`);
+  }
+
+  let jsonArrayR, jsonArrayD;
+
+  // Read both R (numerator) and D (denominator) files
+  [jsonArrayR, jsonArrayD] = await Promise.all(
+    files.map(async (file) => {
+      let csvArray;
+      try {
+        csvArray = await csv().fromFile(file.name);
+      } catch (err) {
+        throw Error(`Error importing weighted metric CSV file ${file.name}`);
+      }
+      try {
+        if (file.format === "v2") {
+          return v2CsvToJson(csvArray);
+        }
+        return v1CsvToJson(csvArray);
+      } catch (err) {
+        throw Error(`Error converting weighted metric CSV to JSON ${file.name}`);
+      }
+    })
+  );
+
+  try {
+    // Divide each value in R by D
+    Object.keys(jsonArrayR).forEach((key) => {
+      Object.keys(jsonArrayR[key]).forEach((key2) => {
+        if (isNumeric(jsonArrayR[key][key2]) && isNumeric(jsonArrayD[key][key2])) {
+          jsonArrayR[key][key2] /= jsonArrayD[key][key2];
+          if (metric.suffix === "%") {
+            jsonArrayR[key][key2] *= 100;
+          }
+        } else {
+          jsonArrayR[key][key2] = null;
+        }
+      });
+    });
+  } catch (err) {
+    throw Error(`Error computing weighted metric value`);
+  }
+  outJSON.w = jsonArrayD;
+  outJSON.map = jsonArrayR;
+  if (metric.accuracy) {
+    outJSON.a = await convertAccuracy(geography, metric, { inputBase });
+  }
+  // Write metric file
+  const dest = path.join(outputBase, geography);
+  return writeMetricFile(dest, metric, outJSON);
+}
+
+/**
+ * Convert given metric from CSV to JSON and write out resulting file(s).
+ *
+ * Can throw errors.
+ */
+async function convertMetric(geography, metric, { inputBase, outputBase }) {
+  if (metric.type === "sum" || metric.type === "mean") {
+    await convertSumOrMeanMetric(geography, metric, { inputBase, outputBase });
+  }
+
+  if (metric.type === "weighted") {
+    await convertWeightedMetric(geography, metric, { inputBase, outputBase });
+  }
+}
+
+/**
+ * Process metrics defined in the data config.
+ *
+ * Never throws, all errors are caught.
+ *
+ * @param {Object} options
+ * @param {string} options.inputBase The base path from which input metrics will be read
```

Comment:
This is super-nitpicky, but just for future readability, the names `inputBase` and `outputBase` make me think of databases -- could they be renamed globally to `inputFileBasePath` and `outputFileBasePath` or something similar?